modules/cli/src/main/scala/scala/cli/commands/package0/Package.scala

```diff
@@ -1016,9 +1073,24 @@ object Package extends ScalaCommand[PackageOptions] with BuildCommandHelpers {
         Nil
     val pythonCliOptions = pythonLdFlags.flatMap(f => Seq("--linking-option", f)).toList
 
+    val libraryLinkingOptions: Seq[String] =
+      Option.when(targetType != PackageType.Native.Application) {
+        /* If we are building a library, we make sure to change the name
+         that the linker will put into the loading path - otherwise
+         the built library will depend on some internal path within .scala-build
+         */
+
+        destPath.flatMap(_.lastOpt).toSeq.flatMap { filename =>
+          Seq("--linking-option", s"-Wl,-install_name,$filename")
+        }
```

Comment:
@WojciechMazur let me know if you have thoughts here - without it the libraries produced depend on themselves (located in .scala-build) so they're not portable

We're doing "convention over configuration" here for better user experience.

Reply:
I've never actually used this one, but seems to be required. I'm fine with with adding this flag, and wonder should we backport it to the ScalaNative toolchain

Reply:
I think ScalaNative toolchain has same issue:

```
$ otool -L target/scala-3.3.1/libscala-native-seed-project-out.dylib
target/scala-3.3.1/libscala-native-seed-project-out.dylib:
        /private/var/folders/rv/tdh6r2xx2b7by3z590lq8vrm0000gn/T/tmp.iWqfQcKp/scala-native-seed-project/target/scala-3.3.1/libscala-native-seed-project-out.dylib
```

Basically this dylib is not portable - it references itself in the target folder